loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+import { Nullable } from "babylonjs/types";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Mesh } from 'babylonjs/Meshes/mesh';
+import { INode, IAccessor } from "../glTFLoaderInterfaces";
+import { IGLTFLoaderExtension } from "../glTFLoaderExtension";
+import { GLTFLoader, ArrayItem } from "../glTFLoader";
+import { Vector3, Quaternion } from 'babylonjs/Maths/math.vector';
+
+const NAME = "KHR_mesh_instancing";
+
+interface IKHRMeshInstancing {
+    mesh?: number;
+    attributes: { [name: string]: number };
+}
+
+/**
+ * [Proposed Specification](https://github.com/KhronosGroup/glTF/pull/1691)
+ * [Playground Sample](//TODO)
+ * !!! Experimental Extension Subject to Changes !!!
+ */
+export class KHR_mesh_instancing implements IGLTFLoaderExtension {
+    /**
+     * The name of this extension.
+     */
+    public readonly name = NAME;
+
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /** @hidden */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public loadNodeAsync(context: string, node: INode, assign: (babylonTransformNode: TransformNode) => void): Nullable<Promise<TransformNode>> {
+        return GLTFLoader.LoadExtensionAsync<IKHRMeshInstancing, TransformNode>(context, node, this.name, (extensionContext, extension) => {
+            const nodeMesh = node.mesh;
```

Comment:
what is `nodeMesh` used for? There is only one instance inside of the loadNodeAsync continuation. It looks like it's being restored at the end, but no one seems to be overriding `node.mesh`?

Reply:
I think my first pass was a bit confused. I assumed I wanted to save nodeMesh as our instance fallback mesh in case the extension did not specify a mesh to instance. The spec has evolved since my first pass, and this is no longer needed. we can safely assume that the mesh that contains the extension is the one to be instanced.